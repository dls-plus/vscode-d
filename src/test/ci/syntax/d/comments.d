// dfmt off
alias baselineCases = AliasSeq!(
    Code!(ubyte, 16 / ubyte.sizeof)([
        /* push   rbp                     */ 0x55,
        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
        /* movd   xmm0,edi                */ 0x66, 0x0f, 0x6e, 0xc7,
        /* punpcklbw xmm0,xmm0            */ 0x66, 0x0f, 0x60, 0xc0,
        /* punpcklwd xmm0,xmm0            */ 0x66, 0x0f, 0x61, 0xc0,
        /* pshufd xmm0,xmm0,0x0           */ 0x66, 0x0f, 0x70, 0xc0, 0x00,
        /* pop    rbp                     */ 0x5d,
        /* ret                            */ 0xc3,
    ]),
    Code!(ubyte*, 16 / ubyte.sizeof)([
        /* push   rbp                     */ 0x55,
        /* mov    rbp,rsp                 */ 0x48, 0x8b, 0xec,
        /* movzx  eax,BYTE PTR [rdi]      */ 0x0f, 0xb6, 0x07,
        /* movd   xmm0,eax                */ 0x66, 0x0f, 0x6e, 0xc0,
        /* punpcklbw xmm0,xmm0            */ 0x66, 0x0f, 0x60, 0xc0,
        /* punpcklwd xmm0,xmm0            */ 0x66, 0x0f, 0x61, 0xc0,
        /* pshufd xmm0,xmm0,0x0           */ 0x66, 0x0f, 0x70, 0xc0, 0x00,
        /* pop    rbp                     */ 0x5d,
        /* ret                            */ 0xc3,
    ])
);