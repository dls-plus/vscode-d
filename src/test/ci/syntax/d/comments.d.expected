[{"start":0,"end":11,"scope":"comment.line.d"}]
[{"start":0,"end":5,"scope":"keyword.other.alias.d"},{"start":5,"end":6,"scope":"source.d"},{"start":6,"end":19,"scope":"variable.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":21,"scope":"keyword.operator.equal.alias.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":30,"scope":"variable.d"},{"start":30,"end":31,"scope":"keyword.operator.logical.d"},{"start":31,"end":32,"scope":"source.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":8,"scope":"variable.d"},{"start":8,"end":9,"scope":"keyword.operator.logical.d"},{"start":9,"end":10,"scope":"source.d"},{"start":10,"end":15,"scope":"storage.type.basic-type.d"},{"start":15,"end":16,"scope":"keyword.operator.comma.d"},{"start":16,"end":17,"scope":"source.d"},{"start":17,"end":19,"scope":"constant.numeric.integer.decimal.d"},{"start":19,"end":20,"scope":"source.d"},{"start":20,"end":21,"scope":"keyword.operator.add.d"},{"start":21,"end":22,"scope":"source.d"},{"start":22,"end":27,"scope":"storage.type.basic-type.d"},{"start":27,"end":34,"scope":"variable.d"},{"start":34,"end":35,"scope":"source.d"},{"start":35,"end":36,"scope":"source.d"},{"start":36,"end":37,"scope":"source.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"},{"start":68,"end":69,"scope":"source.d"},{"start":69,"end":73,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":73,"end":74,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":6,"scope":"source.d"},{"start":6,"end":7,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":4,"scope":"source.d"},{"start":4,"end":8,"scope":"variable.d"},{"start":8,"end":9,"scope":"keyword.operator.logical.d"},{"start":9,"end":10,"scope":"source.d"},{"start":10,"end":15,"scope":"storage.type.basic-type.d"},{"start":15,"end":16,"scope":"keyword.operator.add.d"},{"start":16,"end":17,"scope":"keyword.operator.comma.d"},{"start":17,"end":18,"scope":"source.d"},{"start":18,"end":20,"scope":"constant.numeric.integer.decimal.d"},{"start":20,"end":21,"scope":"source.d"},{"start":21,"end":22,"scope":"keyword.operator.add.d"},{"start":22,"end":23,"scope":"source.d"},{"start":23,"end":28,"scope":"storage.type.basic-type.d"},{"start":28,"end":35,"scope":"variable.d"},{"start":35,"end":36,"scope":"source.d"},{"start":36,"end":37,"scope":"source.d"},{"start":37,"end":38,"scope":"source.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"},{"start":50,"end":51,"scope":"source.d"},{"start":51,"end":55,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":55,"end":56,"scope":"keyword.operator.comma.d"},{"start":56,"end":57,"scope":"source.d"},{"start":57,"end":61,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":61,"end":62,"scope":"keyword.operator.comma.d"},{"start":62,"end":63,"scope":"source.d"},{"start":63,"end":67,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":67,"end":68,"scope":"keyword.operator.comma.d"},{"start":68,"end":69,"scope":"source.d"},{"start":69,"end":73,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":73,"end":74,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":8,"scope":"source.d"},{"start":8,"end":10,"scope":"comment.block.begin.d"},{"start":10,"end":42,"scope":"comment.block.content.d"},{"start":42,"end":44,"scope":"comment.block.end.d"},{"start":44,"end":45,"scope":"source.d"},{"start":45,"end":49,"scope":"constant.numeric.integer.hexadecimal.d"},{"start":49,"end":50,"scope":"keyword.operator.comma.d"}]
[{"start":0,"end":7,"scope":"source.d"}]
[{"start":0,"end":1,"scope":"source.d"},{"start":1,"end":3,"scope":"meta.statement.end.d"}]